src/utils/WebUtils.ts

```diff
@@ -19,8 +23,23 @@ const loadScript = async (scriptUrl: string, callbackOnload: CallableFunction =
     });
 
     scriptElement.addEventListener('error', async () => {
-      await callbackError();
-      reject();
+      if (++attempt >= retries) {
+        await callbackError();
+
+        // Reference: https://developer.mozilla.org/en-US/docs/Web/API/Element/error_event
+        return reject(new Error("Resource failed to load, or can't be used."));
```

Comment:
This is more likely a static error message/detail. Is there a possibility of logging actual error/exception along with this. 

E.g., If loading of script fails due to network error (503 or 404), then is there a way to log these in the logs. This way we will be able to get the root cause as well for the failure.